fdbclient/include/fdbclient/BlobMetadataUtils.h

```diff
@@ -56,6 +56,21 @@ struct BlobMetadataDetailsRef {
 			base = StringRef(arena, from.base.get());
 		}
 	}
+
+	explicit BlobMetadataDetailsRef(Arena& ar,
+	                                BlobMetadataDomainId domainId,
+	                                BlobMetadataDomainNameRef domainName,
+	                                Optional<StringRef> base,
+	                                VectorRef<StringRef> partitions,
+	                                int64_t refreshAt,
+	                                int64_t expireAt)
+	  : domainId(domainId), domainName(ar, domainName), partitions(ar, partitions), refreshAt(refreshAt),
+	    expireAt(expireAt) {
+		if (base.present()) {
+			base = StringRef(ar, base.get());
```

Comment:
nit: do you want to attach `arena` to this StringRef?

Reply:
So since this is a *Ref class that's not supposed to own its own memory, we don't want to make this a standalone. But we do want to copy it from the argument into whatever arena this object is being owned _by_